2022 Privacy chapter (#3173)
HTTPArchive/almanac.httparchive.org: src/content/en/2022/privacy.md

---
title: Privacy
description: Privacy chapter of the 2022 Web Almanac covering adoption and impact of online tracking, privacy preference signals and browser initiatives for a privacy-friendlier web.
authors: [tomvangoethem, nrllh]

> reviewers: [iskander-sanchez-rola]

Reviewer: We should add Melissa Knight as a reviewer as well, I see lots of feedback from them in the doc.

{{ figure_markup(
  image="most-common-trackers.png",
  caption="Most common trackers.",
  description="Bar chart showing the prevalence of trackers on web pages. Google Analytics (site analytics) can be found on 65% desktop sites and 60% of mobile sites, Google (advertising) on 51% and 49% respectively, DoubleClick (advertising) on 50% and 46%, Facebook (advertising) on 30% and 28%, Google Adservices (advertising) on 23% and 21%, Google Syndication (advertising) on 12%, WordPress Stats (site analytics) on 6% for both mobile and desktop, Twitter (Social Media) on 6% and 5%, Adobe Audience Manager (advertising) on 5% and 6%, and finally Rubicon (advertising) on 5% for both mobile and desktop sites.",

>   chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=951980374&format=interactive",

Reviewer: This should have a height attribute set, the figure is getting cut off: https://20220926t211726-dot-webalmanac.uk.r.appspot.com/en/2022/privacy#fig-2

<img width="743" alt="image" src="https://user-images.githubusercontent.com/1120896/192383476-5de675e2-97c0-4c51-8321-f5951620fa72.png">

You can get the exact height by going back to the publishing info in Sheets:

<img width="982" alt="image" src="https://user-images.githubusercontent.com/1120896/192383616-e9436942-72d2-4253-86c8-b844cbb3a7fd.png">

Author: Added height attribute for all graphs with non-default height in a39b587705a3d90cbd9ed62ef1c821e162dcaaec

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=1803269356&format=interactive",
  sheets_gid="315730924",
  sql_file="most_common_permissions_policy_directives.sql"
  )
}}

When we look at the directives that are used in the Permissions Policy, we see a similar usage compared to [last year](../2021/privacy), with the exception of the one that's most widely used in 2022, namely `interest-cohort`. This directive can be used to limit the access to the now-defunct FLoC API. Presumably, this can be attributed to the various shortcomings of FLoC (increases fingerprinting surface, reveals potentially sensitive information about users, ...) where website owners took an active step in trying to protect the privacy of their users.

### Referrer Policy

{{ figure_markup(
  caption="The percentage of desktop sites that sets a document-wide referrer policy.",

>   content="11.64%%",

Reviewer: Throughout, we should standardize on two significant figures in stats. For example "0.33%" and "12%" but not rounding to "0%" or showing unnecessary precision like "11.64%".
```suggestion
  content="12%",
```

Author: Standardised all numbers to two significant figures, except when more decimal places were used in the sentence. Used the following regex to detect such cases: `(\d\d|[1-9])\.\d\d%(?!.*\s0\.\d\d%).*$`

---
title: Privacy
description: Privacy chapter of the 2022 Web Almanac covering adoption and impact of online tracking, privacy preference signals and browser initiatives for a privacy-friendlier web.
authors: [tomvangoethem, nrllh]
reviewers: [iskander-sanchez-rola]
analysts: [max-ostapenko, ydimova]
editors: [DesignrKnight]
translators: []
tomvangoethem_bio: Tom Van Goethem recently joined the Chrome Privacy team at Google. Before, Tom was in PhD program with the DistriNet group of the university of Leuven, Belgium. His research interests cover a broad spectrum of topics in the field of web security and privacy, with a primary focus on side-channel attacks. By uncovering threats and proposing mitigations, Tom aims to make the web a nicer place, a tiny bit at a time.
nrllh_bio: Nurullah Demir is a security researcher and PhD Student at <a hreflang="en" href="https://www.internet-sicherheit.de/en/">Institute for Internet Security</a>. His research focuses on robust web security mechanisms and adversarial machine learning.
results: https://docs.google.com/spreadsheets/d/1iJqj3g0VEjpmjzvtX6VLeRehE7LDQGcw6lOadxGxkjk/

> featured_quote: While it may still take a couple of years to finally get there, we are transitioning towards a web that gives users more control over what they want to share with which parties. We can see this convergence on both sides of the spectrum. On the one hand initiated by the website, and on the other hand enforced by the browser.

Reviewer 2: ```suggestion
featured_quote: While it may still take a couple of years to finally get there, we are transitioning towards a web that gives users more control over what they want to share with which parties. We can see this convergence on both sides of the spectrum. On the one hand, initiated by the website, and on the other hand, enforced by the browser.
```

featured_stat_1: 82%
featured_stat_label_1: Sites that include at least one third-party tracker.
featured_stat_2: 11%
featured_stat_label_2: The prevalence of CMP services, an increase of almost 60% compared to last year.
featured_stat_3: 9.53%
featured_stat_label_3: Sites of the top 1,000 that make use of User-Agent Client Hints, a 3x increase over last year.
unedited: true
---

## Introduction

> Whether it is to keep up-to-date with the latest news, stay in touch with friends via online social media, or look for a nice dress or sweater to buy, many of us rely on the web to provide us these services and information with just a couple of clicks. A side-effect of <a hreflang="en" href="https://datareportal.com/reports/digital-2022-global-overview-report">spending almost 7 hours per day on the internet on average</a>, is that a lot of our browsing activities, and thus indirectly our personal interests and data, is captured or shared with a plethora of online web services and companies.

Reviewer 2: ```suggestion
Whether it is to keep up-to-date with the latest news, stay in touch with friends via online social media, or look for a nice dress or sweater to buy, many of us rely on the web to provide us with these services and information with just a couple of clicks. A side-effect of <a hreflang="en" href="https://datareportal.com/reports/digital-2022-global-overview-report">spending almost 7 hours per day on the internet on average</a>, is that a lot of our browsing activities, and thus indirectly our personal interests and data, is captured or shared with a plethora of online web services and companies.
```

> As advertisers try to provide users with ads that are most relevant to them (as these are the ones they are most likely to interact with), they need to resort to third-party tracking to infer the user's interest. In essence, a user's online activities are tracked step by step, providing trackers, in particular those that are most prevalent on the web, with a heap of information, most of which is probably not even relevant to infer the user's interests. On top of that, users are generally not given an adequate choice to opt out of this.

Reviewer 2: ```suggestion
As advertisers try to provide users with ads that are most relevant to them (as these are the ones they are most likely to interact with), they need to resort to third-party tracking to infer the user's interest. In essence, a user's online activities are tracked step by step, providing trackers, in particular those that are most prevalent on the web, with a heap of information, most of which is probably not even relevant to infer the user's interests. On top of that, users are generally not given an adequate choice to opt-out of this.
```

In this chapter, we explore the current state of the web in terms of privacy. We report on the ubiquitousness of third-party tracking, the different services that make up this ecosystem, and how certain parties are trying to circumvent the protective measures (e.g. blocklist-based anti-trackers) that users are employing to protect their privacy. Furthermore, we also look into how websites are trying to enhance the privacy of their visitors, either by adopting features that limit the information shared with other parties, or being compliant with privacy regulations such as <a hreflang="en" href="https://gdpr.eu/">GDPR</a> and <a hreflang="en" href="https://oag.ca.gov/privacy/ccpa">CCPA</a>.

## Online tracking

{{ figure_markup(
  caption="The percentage of websites that include at least one third-party tracker on desktop.",
  content="82%",
  classes="big-number",
  sheets_gid="225736044",
  sql_file="most_common_tracker_categories.sql"
)
}}

> Tracking is one of the most pervasive web technologies on the web—we find that 82% of desktop websites (80% for mobile) include at least one third-party tracker. By following users' behavior online, these tracking companies can create profiles of them, which can be used for personalized advertising, give insights to website owners on who visits their websites, or use this information to distinguish legitimate users from (unwanted) bots. In this section we explore the different techniques that are used to track the activities of users online and look at how trackers aim to circumvent the various privacy features that aim to protect users from being tracked.

Reviewer 2: ```suggestion
Tracking is one of the most pervasive web technologies on the web—we find that 82% of desktop websites (80% for mobile) include at least one third-party tracker. By following users' behavior online, these tracking companies can create profiles of them, which can be used for personalized advertising, give insights to website owners on who visits their websites, or use this information to distinguish legitimate users from (unwanted) bots. In this section, we explore the different techniques that are used to track the activities of users online and look at how trackers aim to circumvent the various privacy features that aim to protect users from being tracked.
```

### Third-party tracking

> One of the most common forms of online tracking is through third-party services, where a website owner typically includes a third-party (cross-site) script that provides site analytics or shows advertisements to the visitors. This script can then set a third-party cookie, and register which website the user visited. Whenever the user visits another website that includes the same third-party service, the cookie will be sent along to the tracker, allowing them to re-identify the user and link both website visits to the same profile.

Reviewer 2: ```suggestion
One of the most common forms of online tracking is through third-party services, where a website owner typically includes a third-party (cross-site) script that provides site analytics or shows advertisements to visitors. This script can then set a third-party cookie, and register which website the user visited. Whenever the user visits another website that includes the same third-party service, the cookie will be sent along to the tracker, allowing them to re-identify the user and link both website visits to the same profile.
```

> One of the most common forms of online tracking is through third-party services, where a website owner typically includes a third-party (cross-site) script that provides site analytics or shows advertisements to the visitors. This script can then set a third-party cookie, and register which website the user visited. Whenever the user visits another website that includes the same third-party service, the cookie will be sent along to the tracker, allowing them to re-identify the user and link both website visits to the same profile.

Author: Changed to "log which website the user visited" to avoid confusion.

> The types of third-party services that are included, and by doing so are implicitly given the capabilities of tracking website visitors, somewhat vary. The two most common categories (<a hreflang="en" href="https://whotracks.me/blog/tracker_categories.html">as defined by WhoTracks.me</a>) of such trackers are site analytics scripts (68% on mobile, 73% on desktop) and advertising (66% on mobile, 68% on desktop). These two are followed by a couple other categories, some of which might not have a clear link to tracking: customer interaction (services that allow customers to easily send messages to the website owner), audio/video players (e.g. YouTube embedded videos), and social (e.g. Facebook like buttons).

Reviewer 2: ```suggestion
The types of third-party services that are included, and by doing so are implicitly given the capabilities of tracking website visitors, somewhat vary. The two most common categories (<a hreflang="en" href="https://whotracks.me/blog/tracker_categories.html">as defined by WhoTracks.me</a>) of such trackers are site analytics scripts (68% on mobile, 73% on desktop) and advertising (66% on mobile, 68% on desktop). These two are followed by a couple of other categories, some of which might not have a clear link to tracking: customer interaction (services that allow customers to easily send messages to the website owner), audio/video players (e.g. YouTube embedded videos), and social (e.g. Facebook like buttons).
```

{{ figure_markup(
  image="most-common-trackers.png",
  caption="Most common trackers.",

>   description="Bar chart showing the prevalence of trackers on web pages. Google Analytics (site analytics) can be found on 65% desktop sites and 60% of mobile sites, Google (advertising) on 51% and 49% respectively, DoubleClick (advertising) on 50% and 46%, Facebook (advertising) on 30% and 28%, Google Adservices (advertising) on 23% and 21%, Google Syndication (advertising) on 12%, WordPress Stats (site analytics) on 6% for both mobile and desktop, Twitter (Social Media) on 6% and 5%, Adobe Audience Manager (advertising) on 5% and 6%, and finally Rubicon (advertising) on 5% for both mobile and desktop sites.",

Reviewer 2: ```suggestion
  description="Bar chart showing the prevalence of trackers on web pages. Google Analytics (site analytics) can be found on 65% of desktop sites and 60% of mobile sites, Google (advertising) on 51% and 49% respectively, DoubleClick (advertising) on 50% and 46%, Facebook (advertising) on 30% and 28%, Google Adservices (advertising) on 23% and 21%, Google Syndication (advertising) on 12%, WordPress Stats (site analytics) on 6% for both mobile and desktop, Twitter (Social Media) on 6% and 5%, Adobe Audience Manager (advertising) on 5% and 6%, and finally Rubicon (advertising) on 5% for both mobile and desktop sites.",
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=1803959066&format=interactive",
  sheets_gid="1992611959",
  sql_file="number_of_websites_with_nb_trackers.sql"
  )
}}

Websites might want to make use of multiple third-party services, and thus may include multiple trackers in their website (be sure to check out the [third parties chapter](./third-parties) for a deep dive into which third-parties are included on the web!). We find that approximately 15% of desktop sites and 16% of mobile sites include "just" one tracker. Unfortunately, this means that it is in fact more common for websites to include multiple trackers. We even found one website that included 126 different trackers!

### (Re)targeting

{{ figure_markup(
  image="retargeting-services.png",
  caption="Most common retargeting services.",

>   description="Bar chart showing the percentage of pages including a particular retargeting service. Criteo was found on 2.04% desktop and 1.98% mobile sites respectively, Yahoo Advertising on 0.44% and 0.54%, AdRoll on 0.34% and 0.49%, OptiMonk on 0.09% and 0.11%, SharpSpring Ads on 0.08% and 0.12%. The remaining entries in the chart have a prevalence of less then 0.10%. These are Albacross, Smarter Click, Blue, SteelHouse, Cross Pixel, Linx Impulse, and Picreel.",

Reviewer 2: ```suggestion
  description="Bar chart showing the percentage of pages including a particular retargeting service. Criteo was found on 2.04% desktop and 1.98% mobile sites respectively, Yahoo Advertising on 0.44% and 0.54%, AdRoll on 0.34% and 0.49%, OptiMonk on 0.09% and 0.11%, SharpSpring Ads on 0.08% and 0.12%. The remaining entries in the chart have a prevalence of less than 0.10%. These are Albacross, Smarter Click, Blue, SteelHouse, Cross Pixel, Linx Impulse, and Picreel.",
```

  sheets_gid="96406513",
  sql_file="number_of_websites_using_each_retargeting.sql"
  )
}}

When browsing the web, we often encounter advertisements for products that we recently looked up. The reason for that is ad retargeting. When a website detects that a user might be interested in a specific product, they report this to the tracker and/or advertiser, who will later on, when the user is visiting other, unrelated websites, show advertisements for the product that the user is supposedly interested in, in an attempt to nudge them into purchasing it.
The tracker offering most of the purely retargeting services is Criteo, with a prevalence of 1.98% on desktop and 2.04% on mobile. It is followed by Yahoo Advertising and AdRoll, which collectively make up less than half of Criteo's market share. The most widely used retargeting service of [last year](../2021/privacy), Google Tag Manager, does not show in these results as it is now classified under the "tag managers" Wappalyzer category. Although this service still offers retargeting, it also provides a larger variety of other services.

### Third-party cookies

> As mentioned before, the most established way to track users across different websites is by means of third-party cookies. With recent changes in browser policies, cookies will no longer be included in cross-site requests by default. In technical terms this means that most browsers set the `SameSite` attribute of cookies to the default value `Lax`. Websites can override this by explicitly setting the value themselves. This has been happening on a large scale: of the third-party cookies that set the SameSite cookies, 98% of them set it to the value `None`, allowing them to be included in cross-site requests. Furthermore, the expiration time of the cookie also determines how long it remains valid; we find that the median lifetime of a cookie is 365 days. For a deeper dive into cookies and cookie attributes, please refer to the [security chapter](./security).

Reviewer 2: ```suggestion
As mentioned before, the most established way to track users across different websites is by means of third-party cookies. With recent changes in browser policies, cookies will no longer be included in cross-site requests by default. In technical terms, this means that most browsers set the `SameSite` attribute of cookies to the default value `Lax`. Websites can override this by explicitly setting the value themselves. This has been happening on a large scale: of the third-party cookies that set the SameSite cookies, 98% of them set it to the value `None`, allowing them to be included in cross-site requests. Furthermore, the expiration time of the cookie also determines how long it remains valid; we find that the median lifetime of a cookie is 365 days. For a deeper dive into cookies and cookie attributes, please refer to the [security chapter](./security).
```

{{ figure_markup(
  image="cookie-origins.png",
  caption="Top 10 origins of cookies set by trackers.",

>   description="Bar chart showing the prevalence of the different origins that set cross-site cookies. Cookies set by doubleclick.net can be found on 26.34% desktop and 24.06% mobile sites respectively, for facebook.com this was 18.86% for mobile and 17.48% for desktop, cookies set by youtube.com can be found on 9.65% mobile and 8.65% desktop sites, google.com on 6.40% and 5.98%, yandex.ru on 4.32% and 4.91%, linkedin.com on 4.09% and 3.19%, adsrvr.org on 4.02% and 3.77%, pubmatic.com on 4.00% and 4.88%, yahoo.com on 3.94% and 3.62%, bing.com on 3.76% and 2.81%, and finally cookies were set by rubiconproject.com on 3.73% desktop and 3.30% mobile sites.",

Reviewer 2: ```suggestion
  description="Bar chart showing the prevalence of the different origins that set cross-site cookies. Cookies set by doubleclick.net can be found on 26.34% of desktop and 24.06% of mobile sites respectively, for facebook.com this was 18.86% for mobile and 17.48% for desktop, cookies set by youtube.com can be found on 9.65% mobile and 8.65% desktop sites, google.com on 6.40% and 5.98%, yandex.ru on 4.32% and 4.91%, linkedin.com on 4.09% and 3.19%, adsrvr.org on 4.02% and 3.77%, pubmatic.com on 4.00% and 4.88%, yahoo.com on 3.94% and 3.62%, bing.com on 3.76% and 2.81%, and finally cookies were set by rubiconproject.com on 3.73% desktop and 3.30% mobile sites.",
```

{{ figure_markup(
  image="most-common-cookies.png",
  caption="Top 10 cookies set by trackers.",
  description="Bar chart showing the prevalence of specific cross-site cookies, grouped by the cookie name. The cookie named `test_cookie`, which was set by doubleclick.net can be found on 26% desktop sites and 24% mobile sites respectively, `fr` set by facebook.com was found on 19% and 17%, `IDE` set by doubleclick.net on 12% and 12%, `YSC` set by youtube.com on 10% and 9%, `VISITOR_INFO1_LIVE` set by youtube.com on 10% and 8%, `sync_cookie_csrf` set by yandex.ru on 4% and 5%, `yandexuid` set by yandex.com on 4% and 5%, `yuidss` set by yandex.com on 4% and 5%, `i` set by yandex.com on 4% and 5%, and finally `ymex` set by yandex.com on 4% and 5%.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=1506966442&format=interactive",
  sheets_gid="1112448573",
  sql_file="top100_cookies_set_from_header.sql"
  )
}}

> When looking at the most common third-party cookies, we again see several tracking domains, lead by the `test_cookie` from doubleclick.net (a cookie with a lifespan of 15 minutes that is used for functionality purposes according to <a hreflang="en" href="https://business.safety.google/adscookies/">its description</a>). This cookie is followed by the `fr` cookie set by facebook.com, a cookie "used to deliver, measure and improve the relevancy of ads, with a lifespan of 90 days" according to <a hreflang="en" href="https://www.facebook.com/policy/cookies/">its definition</a>. The rest of the 10 most prevalent third-party cookies are set by YouTube and Yandex.

Reviewer 2: ```suggestion
When looking at the most common third-party cookies, we again see several tracking domains, led by the `test_cookie` from doubleclick.net (a cookie with a lifespan of 15 minutes that is used for functionality purposes according to <a hreflang="en" href="https://business.safety.google/adscookies/">its description</a>). This cookie is followed by the `fr` cookie set by facebook.com, a cookie "used to deliver, measure and improve the relevancy of ads, with a lifespan of 90 days" according to <a hreflang="en" href="https://www.facebook.com/policy/cookies/">its definition</a>. The rest of the 10 most prevalent third-party cookies are set by YouTube and Yandex.
```

### Evasion technique: fingerprinting

{{ figure_markup(
  image="fingerprinting-services.png",
  caption="Fingerprinting services usage.",
  description="Bar chart showing the prevalence of fingerprinting services. The fingerprinting script FingerprintJS was found on 0.62% desktop and 0.73% mobile sites respectively, ClientJS was detected on 0.04% and 0.04%, MaxMind on 0.03% and 0.04%, and finally TruValidate on 0.02% and 0.03%.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=79170171&format=interactive",
  sheets_gid="1069316937",
  sql_file="number_of_websites_using_each_fingerprinting.sql"
  )
}}

> As more and more browsers develop countermeasures for cookie-based tracking, and giving users more control to block third-party cookies, some trackers aim to circumvent these protections. One such technique is fingerprinting, where browser-specific (e.g. installed browser extensions), OS-specific (e.g. installed fonts) and hardware-specific (e.g. differences in rendering complex composition based on which GPU is used) features are used to create a unique fingerprint of the user. This fingerprint then allows the tracker to re-identify the same user across different, unrelated websites.

Reviewer 2: ```suggestion
As more and more browsers develop countermeasures for cookie-based tracking, and give users more control to block third-party cookies, some trackers aim to circumvent these protections. One such technique is fingerprinting, where browser-specific (e.g. installed browser extensions), OS-specific (e.g. installed fonts), and hardware-specific (e.g. differences in rendering complex composition based on which GPU is used) features are used to create a unique fingerprint of the user. This fingerprint then allows the tracker to re-identify the same user across different, unrelated websites.
```

{{ figure_markup(
  image="cname-tracking-services.png",
  caption="Top 10 CNAME tracking services.",
  description="Bar chart showing the prevalence of CNAME trackers. Adobe Experience Cloud was found on 0.65% desktop and 0.38% mobile sites respectively, Pardot on 0.44% and 0.25%, Oracle Eloqua on 0.06% and 0.03%, Act-On Software on 0.05% and 0.03%, Webtrekk on 0.02% and 0.01%, and Eulerian on 0.01% and 0.01%. The remaining CNAME trackers are shown with 0.00% prevalence. These are AT Internet, Tracedock, Ingenious Technologies, Criteo, Wizaly, and Keyade.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=118406352&format=interactive",
  sheets_gid="1717363829",
  sql_file="nb_sites_with_cname_tracking.sql"
  )
}}

> By analyzing the various CNAME trackers, we find that the market share is mainly concentrated around two main services: Adobe Experience Cloud (0.65% on desktop and 0.38% on mobile) and Pardot (0.25% on desktop and 0.44% on mobile). Interestingly, the adoption of CNAME tracking is significantly higher on websites visited with a desktop browser compared to those visited on mobile. Presumably this is because there are fewer privacy-preserving mechanisms on mobile browsers (e.g. most of the popular browsers on mobile do not support extensions).

Reviewer 2: ```suggestion
By analyzing the various CNAME trackers, we find that the market share is mainly concentrated around two main services: Adobe Experience Cloud (0.65% on desktop and 0.38% on mobile) and Pardot (0.25% on desktop and 0.44% on mobile). Interestingly, the adoption of CNAME tracking is significantly higher on websites visited with a desktop browser compared to those visited on mobile. Presumably, this is because there are fewer privacy-preserving mechanisms on mobile browsers (e.g. most of the popular browsers on mobile do not support extensions).
```

{{ figure_markup(
  image="cname-tracking-by-rank.png",
  caption="CNAME tracking usage per website rank group.",

>   description="Bar chart showing the prevalence of CNAME trackers for different rank groups ranging from 1,000 until 1,000,000 and a separate category that capturs all sites. The top 1,000 sites has CNAME tracking on 6.2% desktop and 5.8% mobile sites respectively, the top 10,000 has 5.9% and 5.3%, the top 100,000 has 2.9% and 2.7%, the top 1,000,000 has 1.3% and 1.2%, and finally the overall prevalence of CNAME tracking is 0.9% on desktop and 0.5% on mobile sites.",

Reviewer 2: ```suggestion
  description="Bar chart showing the prevalence of CNAME trackers for different rank groups ranging from 1,000 to 1,000,000 and a separate category that captures all sites. The top 1,000 sites has CNAME tracking on 6.2% desktop and 5.8% mobile sites respectively, the top 10,000 has 5.9% and 5.3%, the top 100,000 has 2.9% and 2.7%, the top 1,000,000 has 1.3% and 1.2%, and finally the overall prevalence of CNAME tracking is 0.9% on desktop and 0.5% on mobile sites.",
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=2114701877&format=interactive",
  sheets_gid="217371442",
  sql_file="number_of_websites_with_device_sensor_events.sql"
  )
}}

We find that the most prominent sensor event that websites listen for is the `deviceOrientation` event, which fires when the device changes from portrait to landscape mode or vice versa. It is used on 4.04% of desktop websites and 4.10% of mobile websites. The usage is likely this high (relatively) because websites might want to update elements of the layout when the orientation of the device changes.

### Media devices

{{ figure_markup(
  caption="The percentage of desktop pages that enumerate media devices.",
  content="0.59%",
  classes="big-number",
  sheets_gid="1554147968",
  sql_file="number_of_websites_with_device_sensor_blink_usage.sql"
)
}}

> Using the [MediaDevices API](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices), web developers can use the `enumerateDevices()` method to get a list of all media devices connected to the user's device. While this feature is useful to determine whether a user has a camera or microphone connected to initiate a video call, it can also be used to gather information about the system's environment for fingerprinting purposes. We find that 0.59% desktop websites and 0.48% mobile sites try to access the list of connected media devices (note that our crawler does not interact with the site, nor click on any buttons). Interestingly, the usage of this API has significantly reduced since last year, when **7.23%** of desktop sites were accessing the list of media devices. Most likely this is due to a popular library that no longer calls the API.

Reviewer 2: ```suggestion
Using the [MediaDevices API](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices), web developers can use the `enumerateDevices()` method to get a list of all media devices connected to the user's device. While this feature is useful to determine whether a user has a camera or microphone connected to initiate a video call, it can also be used to gather information about the system's environment for fingerprinting purposes. We find that 0.59% of desktop websites and 0.48% of mobile sites try to access the list of connected media devices (note that our crawler does not interact with the site, nor click on any buttons). Interestingly, the usage of this API has significantly reduced since last year, when **7.23%** of desktop sites were accessing the list of media devices. Most likely this is due to a popular library that no longer calls the API.
```

## Established controls to improve visitor's privacy

As websites include a lot of content (scripts, plugins, ...) from third-parties that they might not entirely trust, they might want to protect their users' privacy from these third-parties. Next, we explore the various controls that can be used to restrict the features or data that third-parties have access to, or that make it explicitly clear which information a website wants to obtain from a user.

### Permission Policy

{{ figure_markup(
  image="permissions-policy-type.png",
  caption="Permissions Policy usage by API type.",

>   description="Bar chart showing which percentage of websites use permission policy, either the Permissions Policy or Feature Policy. The Feature Policy API was used on 0.69% desktop and 0.52% mobile sites respectively, the Permissions Policy API was found on 2.71% desktop and 2.31% mobile sites and in total 3.26% desktop and 2.73% mobile sites controlled which features their site supported by setting a policy.",

Reviewer 2: ```suggestion
  description="Bar chart showing which percentage of websites use permission policy, either the Permissions Policy or Feature Policy. The Feature Policy API was used on 0.69% of desktop and 0.52% of mobile sites respectively, the Permissions Policy API was found on 2.71% of desktop and 2.31% of mobile sites, and in total 3.26% of desktop and 2.73% of mobile sites controlled which features their site supported by setting a policy.",
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=1921531833&format=interactive",
  sheets_gid="741173570",
  sql_file="number_of_websites_with_permissions_policy.sql"
  )
}}

> By default, any (third-party) script can access the same browser features as the website they're embedded in. In order to limit the features that will be enabled for the website, the website can make use of the <a hreflang="en" href="https://developer.chrome.com/en/docs/privacy-sandbox/permissions-policy/">Permissions Policy</a>. Through an HTTP response header the website can indicate which features it wants enabled. For instance, if the `microphone` feature is not included in this list, none of the scripts embedded in the web page can use it. Although the policy is fairly new, we are seeing an adoption of 2.71% on desktop sites and 2.31% on mobile sites.

Reviewer 2: ```suggestion
By default, any (third-party) script can access the same browser features as the website they're embedded in. In order to limit the features that will be enabled for the website, the website can make use of the <a hreflang="en" href="https://developer.chrome.com/en/docs/privacy-sandbox/permissions-policy/">Permissions Policy</a>. Through an HTTP response header the website can indicate which features it wants to be enabled. For instance, if the `microphone` feature is not included in this list, none of the scripts embedded in the web page can use it. Although the policy is fairly new, we are seeing an adoption of 2.71% on desktop sites and 2.31% on mobile sites.
```

The Permissions Policy supersedes the [Feature Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy), which can still be found on 0.69% of desktop sites and 0.52% of mobile sites.
By default most of the features regulated by the Permissions Policy are disabled in cross-origin iframes, they can be explicitly enabled through the `allow` attribute. We find that 15.18% of desktop sites and 14.32% of mobile sites make use of this feature. For a more detailed analysis on the use of the `allow` attribute on iframes, please refer to [the security chapter](./security).

> {{ figure_markup(

Reviewer 2: Image seems to cut off
![image](https://user-images.githubusercontent.com/27865704/192446457-9a2585fe-356a-4927-bafe-fe50ebfe85c0.png)

  image="permission-policy-features.png",
  caption="Most common Permissions Policy feature names.",

>   description="Bar chart showing which Permissions Policy features are most prevalent. The feature `interest-cohort` of the Permissions Policy was present on 1.18% desktop and 0.93% mobile sites respectively, the `geolocation` feature was specificied on 0.80% and 0.58% sites, `microphone` on 0.78% and 0.57%, `camera` on 0.75% and 0.55%, `payment` on 0.57% and 0.42%, `gyroscope` on 0.54% and 0.40%, `magnetometer` on 0.54% and 0.39%, `fullscreen` on 0.47% and 0.34%, `usb` on 0.45% and 0.33%, and finally `accelerometer` was defined on 0.44% desktop and 0.32% mobile sites.",

Reviewer 2: ```suggestion
  description="Bar chart showing which Permissions Policy features are most prevalent. The feature `interest-cohort` of the Permissions Policy was present on 1.18% desktop and 0.93% mobile sites respectively, the `geolocation` feature was specified on 0.80% and 0.58% sites, `microphone` on 0.78% and 0.57%, `camera` on 0.75% and 0.55%, `payment` on 0.57% and 0.42%, `gyroscope` on 0.54% and 0.40%, `magnetometer` on 0.54% and 0.39%, `fullscreen` on 0.47% and 0.34%, `usb` on 0.45% and 0.33%, and finally `accelerometer` was defined on 0.44% desktop and 0.32% mobile sites.",
```

### User-Agent Client Hints

In an effort to reduce the information that is revealed about the browser environment, and more specifically the User-Agent string, the <a hreflang="en" href="https://wicg.github.io/ua-client-hints/">User-Agent Client Hints</a> mechanism was introduced. Through this feature, websites that want to access certain information about the user's browsing environment (browser version, operating system, ...) now have to set a header (`Accept-CH`) in the first response, upon which the browser will send the requested data in subsequent requests. Among other benefits, this feature reduces the fingerprinting surface and allows browsers to intervene in sending certain data, e.g. via the <a hreflang="en" href="https://github.com/mikewest/privacy-budget">Privacy Budget</a> proposal).

{{ figure_markup(
  image="client-hints-by-rank.png",
  caption="Number of websites with Client Hints by rank group.",

>   description="Bar chart showing the prevalence of websites employing User-Agent Client Hints for different rank groups ranging from 1,000 until 1,000,000 and a separate category that capturs all sites. The top 1,000 sites enabled Client Hints on 9.53% desktop and 9.11% mobile sites respectively, on the top 10,000 Client Hints was found on 3.14% desktop and 3.12% mobile sites, on the top 100,000 on 1.02% and 1.05%, on the top 1,000,000 on 0.38% and 0.39%, and overall Client Hints was used on 0.31% desktop and 0.56% mobile sites.",

Reviewer 2: ```suggestion
  description="Bar chart showing the prevalence of websites employing User-Agent Client Hints for different rank groups ranging from 1,000 to 1,000,000 and a separate category that captures all sites. The top 1,000 sites enabled Client Hints on 9.53% desktop and 9.11% mobile sites respectively, on the top 10,000 Client Hints was found on 3.14% desktop and 3.12% mobile sites, on the top 100,000 on 1.02% and 1.05%, on the top 1,000,000 on 0.38% and 0.39%, and overall Client Hints was used on 0.31% desktop and 0.56% mobile sites.",
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=318395619&format=interactive",
  sheets_gid="1199573159",
  sql_file="number_of_websites_with_client_hints.sql"
  )
}}

> When we look at the adoption of sites that respond with the `Accept-CH` header in comparison with the results from [last year](../2021/privacy) (top 1k: 3.56%, top 10k: 1.44%), we see a significant increase in adoption, almost 3x for the most popular sites. Presumably this increase in adoption is related to the fact Chrome has been reducing the information that is shared in the User-Agent string (through the <a hreflang="en" href="https://www.chromium.org/updates/ua-reduction/">User-Agent Reduction plan</a>).

Reviewer 2: ```suggestion
When we look at the adoption of sites that respond with the `Accept-CH` header in comparison with the results from [last year](../2021/privacy) (top 1k: 3.56%, top 10k: 1.44%), we see a significant increase in adoption, almost 3x for the most popular sites. Presumably, this increase in adoption is related to the fact Chrome has been reducing the information that is shared in the User-Agent string (through the <a hreflang="en" href="https://www.chromium.org/updates/ua-reduction/">User-Agent Reduction plan</a>).
```

> When we look at the adoption of sites that respond with the `Accept-CH` header in comparison with the results from [last year](../2021/privacy) (top 1k: 3.56%, top 10k: 1.44%), we see a significant increase in adoption, almost 3x for the most popular sites. Presumably this increase in adoption is related to the fact Chrome has been reducing the information that is shared in the User-Agent string (through the <a hreflang="en" href="https://www.chromium.org/updates/ua-reduction/">User-Agent Reduction plan</a>).

Author: Updated to "Chromium".

We find that the sites that make use of User-Agent Client hints, generally request access to a relatively large number of properties, limiting the benefit of what browsers aim to achieve through efforts such as User-Agent Reduction. It will be interesting to see in the near future how/whether browsers will limit the practices of acquiring a lot of information about the user's browsing environment.

## New efforts to improve privacy by the browser

> Over the last few years, the average web user has become increasingly conscious about their online privacy. On the one hand, the many data breaches, which just seem to <a hreflang="en" href="https://www.informationisbeautiful.net/visualizations/worlds-biggest-data-breaches-hacks/">keep on happening and getting bigger and bigger</a>, have left very few unaffected. On the other hand, the fact of the ubiquitous tracking of users through third-party cookies is becoming increasingly well known within the general population. As a result, more and more users are starting to expect their browser to protect their privacy, and give them more control over the tracking online behaviors. Browser vendors, online publishers and ad-tech companies have heard this demand for improved privacy, and have proposed the Privacy Sandbox (an initiative led by Google Chrome).

Reviewer 2: ```suggestion
Over the last few years, the average web user has become increasingly conscious about their online privacy. On the one hand, the many data breaches, which just seem to <a hreflang="en" href="https://www.informationisbeautiful.net/visualizations/worlds-biggest-data-breaches-hacks/">keep on happening and getting bigger and bigger</a>, have left very few unaffected. On the other hand, the fact of the ubiquitous tracking of users through third-party cookies is becoming increasingly well known within the general population. As a result, more and more users are starting to expect their browser to protect their privacy, and give them more control over the tracking of online behaviors. Browser vendors, online publishers, and ad-tech companies have heard this demand for improved privacy, and have proposed the Privacy Sandbox (an initiative led by Google Chrome).
```

### Privacy Sandbox Origin Trial

> At the time of publishing this year's Web Almanac, Privacy Sandbox features are not yet available for general use. Websites and web services (such as ads, which are typically shown in iframes) can however participate in early testing of  the Privacy Sandbox features, by making use of the <a hreflang="en" href="https://developer.chrome.com/en/blog/privacy-sandbox-unified-origin-trial/">Origin Trial</a>. Note that this is only for users whose browser supports the feature (Privacy Sandbox features are only implemented in Chrome, and are still disabled by default at the time of this writing). This gives the web services access to three Privacy Sandbox-related APIs: <a hreflang="en" href="https://developer.chrome.com/docs/privacy-sandbox/topics/">Topics</a>, <a hreflang="en" href="https://developer.chrome.com/docs/privacy-sandbox/fledge/">FLEDGE</a>, and <a hreflang="en" href="https://developer.chrome.com/docs/privacy-sandbox/attribution-reporting/">Attribution Reporting</a>.

Reviewer 2: ```suggestion
At the time of publishing this year's Web Almanac, Privacy Sandbox features are not yet available for general use. Websites and web services (such as ads, which are typically shown in iframes) can however participate in early testing of the Privacy Sandbox features, by making use of the <a hreflang="en" href="https://developer.chrome.com/en/blog/privacy-sandbox-unified-origin-trial/">Origin Trial</a>. Note that this is only for users whose browser supports the feature (Privacy Sandbox features are only implemented in Chrome, and are still disabled by default at the time of this writing). This gives the web services access to three Privacy Sandbox-related APIs: <a hreflang="en" href="https://developer.chrome.com/docs/privacy-sandbox/topics/">Topics</a>, <a hreflang="en" href="https://developer.chrome.com/docs/privacy-sandbox/fledge/">FLEDGE</a>, and <a hreflang="en" href="https://developer.chrome.com/docs/privacy-sandbox/attribution-reporting/">Attribution Reporting</a>.
```

        <td class="numeric">10.51%</td>
      </tr>
      <tr>
        <td>https://googleadservices.com</td>
        <td class="numeric">2.50%</td>
        <td class="numeric"d>2.29%</td>
      </tr>
      <tr>
        <td>https://s.pinimg.com</td>
        <td class="numeric">1.49%</td>
        <td class="numeric"d>1.21%</td>
      </tr>
      <tr>
        <td>https://criteo.net</td>
        <td class="numeric">0.64%</td>
        <td class="numeric"d>0.41%</td>
      </tr>
      <tr>
        <td>https://criteo.com</td>
        <td class="numeric">0.59%</td>
        <td class="numeric"d>0.37%</td>
      </tr>
      <tr>
        <td>https://imasdk.googleapis.com</td>
        <td class="numeric">0.10%</td>
        <td class="numeric"d>0.07%</td>
      </tr>
      <tr>
        <td>https://teads.tv</td>
        <td class="numeric">0.04%</td>
        <td class="numeric"d>0.03%</td>
      </tr>
    </tbody>
  </table>
  <figcaption>
    {{ figure_link(
      caption="Prevalence of origins requesting access to the Privacy Sandbox API Origin Trial.",
      sheets_gid="1410031518",
      sql_file="number_of_websites_with_origin_trial_from_token.sql",
    ) }}
  </figcaption>
</figure>

> The most prevalent services on the web that will test during  the Origin Trial of Privacy Sandbox are: Google Tag Manager, Doubleclick, Google Syndication and Google Ad Services make up the top five on both desktop and mobile sites. These are followed by the social media site Pinterest, and other trackers and advertisers: Criteo, Google Ads SDK and Teads.

Reviewer 2: ```suggestion
The most prevalent services on the web that will test during the Origin Trial of Privacy Sandbox are: Google Tag Manager, Doubleclick, Google Syndication and Google Ad Services make up the top five on both desktop and mobile sites. These are followed by the social media site Pinterest, and other trackers and advertisers: Criteo, Google Ads SDK, and Teads.
```

### IAB consent frameworks

Compared to GDPR, the <a hreflang="en" href="https://iabeurope.eu">IAB Europe Transparency and Consent Framework (TCF)</a> is an industry-standard where <a hreflang="en" href="https://iabeurope.eu/vendor-list/">global vendors</a> are involved. The goal is to establish communication between user consent and advertisers. TCF ensures that the websites in Europe are GDPR-compliant. IAB Tech Lab US developed <a hreflang="en" href="https://iabtechlab.com/standards/ccpa/">the U.S. Privacy Technical Specifications (USP)</a>  was designed for the United States using the same concept of TCF. Below we show websites with IAB. We record that 4.6% of websites use any USP and while 2.2% of websites use IAB. Thus, we have recorded an increase for both specifications since last year. We would like to note here that our measurement is USA-based, so according to TCF, no consent banner is required for non-EU visits. So this can be the reason why we identify more websites with USP.

{{ figure_markup(
  image="iab-prevalence.png",
  caption="Websites with IAB.",

>   description="Bar chart showing the most common IAB frameworks. Overall, IAB was found on 4.6% desktop and 4.4% mobile sites respectively, the IAB USP was found on 3.5% and 3.4% sites, IAB TCF was found on 2.2% and 1.9% of which 2.1% and 1.8% were for version 2, and 0.4% and 0.3% respectively were for version 1.",

Reviewer 2: "IAB USP was found on 3.5% and 3.4% sites"
I think a respectfully is missing
```suggestion
  description="Bar chart showing the most common IAB frameworks. Overall, IAB was found on 4.6% of desktop and 4.4% of mobile sites respectively, the IAB USP was found on 3.5% and 3.4% sites, IAB TCF was found on 2.2% and 1.9% of which 2.1% and 1.8% were for version 2, and 0.4% and 0.3% respectively were for version 1.",
```

  sheets_gid="772029978",
  sql_file="number_of_websites_with_iab.sql"
  )
}}

In the following, we present the top CMP providers for AIB TCF v2.

{{ figure_markup(
  image="iab-tcfv2-prevalence.png",
  caption="Top CMPs for IAB TCF v2.",
  description="Bar chart showing the most common CMP providers for IAB TCF version 2. Quantcast International Limited was the CMP provider on 0.37% desktop and 0.33% mobile sites, Google LLC on 0.34% and 0.29%, Didomi on 0.31% and 0.26%, 1020, Inc. dba Placecast and Ericsson Emodo on 0.23% and 0.17%, iubenda on 0.10% and 0.10%, Sourcepoint Technologies, Inc. on 0.07% and 0.07%, and finally SIRDATA on 0.06% and 0.07% sites.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=1405483323&format=interactive",
  sheets_gid="1374296424",
  sql_file="most_common_cmps_for_iab_tcf_v2.sql"
  )
}}

We see that Quantcast International Limited (0.37%), Google LLC (0.34%) and Didomi (0.31%) are popular CMP providers for IAB TCF v2.

{{ figure_markup(
  image="iab-publisher-countries.png",
  caption="Most common publisher countries in IAB TCF v2.",

>   description="Bar chart showing the most common countries of publishers that employ IAB TCF v2. The country of the publisher was unknown for 0.31% desktop and 0.29% mobile sites, on 0.04% sites the country of the publisher was set to Germnary, on 0.04% desktop and 0.03% mobile sites it was set to United States, on 0.03% and 0.02% European Union, and finally on 0.02% and 0.01% sites it was United Kingdom.",

Reviewer 2: ```suggestion
  description="Bar chart showing the most common countries of publishers that employ IAB TCF v2. The country of the publisher was unknown for 0.31% of desktop and 0.29% of mobile sites, on 0.04% of sites the country of the publisher was set to Germany, on 0.04% of desktop and 0.03% of mobile sites it was set to the United States, on 0.03% and 0.02% European Union, and finally, on 0.02% and 0.01% sites it was the United Kingdom.",
```

  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=1828110274&format=interactive",
  sheets_gid="1272054750",
  sql_file="most_common_countries_for_iab_tcf_v2.sql"
  )
}}

Our analysis shows that the most common publishers we identified are from Germany, the US, and the EU.

### Privacy policy

> Notifications regarding data processing do not always take place via a consent banner. They are also usually described in more detail on separate pages compared to such banners. On such pages, you will find information on integrated 3rd parties, which data is processed for which purpose, etc. To identify such sites, we used the privacy-relevant signatures from <a hreflang="en" href="https://github.com/RUB-SysSec/we-value-your-privacy/blob/master/privacy_wording.json">a study</a>. Using this method, we could determine that 45% of the websites (mobile: 41%)  contained a link on their homepage to a privacy-related page. The figure below shows the distribution of the top privacy link keywords.

Reviewer 2: We have used third-party across the chapter, so 3rd parties can be reworded

{{ figure_markup(
  image="privacy-link-keywords.png",
  caption="Top privacy link keywords.",
  description="Bar chart showing the most common keywords used to refer to the website's privacy policy. The word 'privacy' was used on 28.63% desktop and 22.95% mobile sites to link to the privacy policy, the word 'policy' on 24.26% and 19.41% sites, 'cookies' on 8.19% and 7.90%, 'cookie policy' on 3.63% and 3.30%, 'privacidad' on 2.68% and 2.99%, 'datenschutz' on 2.09% and 3.14%, 'mentions légales' on 2.08% and 1.85%, 'privacidade' on 1.66% and 1.76%, 'aviso legal' on 1.35% and 1.65%, 'prywatności' on 0.97% and 1.12%, and finally the word 'gdpr' was used on 0.99% and 0.94% sites to link to the privacy policy.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=817391735&format=interactive",
  sheets_gid="1284713488",
  sql_file="most_common_privacy_link_keywords.sql"
  )
}}

We see that privacy (29%), policy (24%), and cookies (8%) are the top keywords for such links.

## Conclusion

> In this chapter we explored many different aspects related to our online privacy on the web. It is clear that in the past year a considerable amount of things have changed that affect our privacy, and this progress can be expected to continue in the following years. In short, there are some exciting times ahead of us. On the one hand, we found some unfortunate evolutions, which hopefully one day we will be able to refer to as the web's legacy. Third-party tracking, mainly fueled by third-party cookies, is still ubiquitous with over 82% of websites containing at least a single tracker. Furthermore, there still is a non-negligible number of websites or web services that employ evasive techniques to circumvent anti-tracking measures.

Reviewer 2: ```suggestion
In this chapter, we explored many different aspects related to our online privacy on the web. It is clear that in the past year a considerable amount of things have changed that affect our privacy, and this progress can be expected to continue in the following years. In short, there are some exciting times ahead of us. On the one hand, we found some unfortunate evolutions, which hopefully one day we will be able to refer to as the web's legacy. Third-party tracking, mainly fueled by third-party cookies, is still ubiquitous with over 82% of websites containing at least a single tracker. Furthermore, there still is a non-negligible number of websites or web services that employ evasive techniques to circumvent anti-tracking measures.
```

On a more positive, privacy-preserving track, we find that fewer sites are trying to access potentially sensitive information from browser APIs. Hopefully this remains the case with the new APIs that are introduced in browsers on a regular basis.

> Generally, it seems that websites are starting to hear the call of users to respect their privacy (a call that is getting louder and louder). More and more sites are switching to employing browser features that restrict the information that is sent to third parties. Furthermore, mainly motivated by privacy regulations such as GDPR and CCPA, we are seeing a clear increase (of almost 60%) in the adoption of consent management platforms (CMPs), giving users more control on which information they want to share.

Reviewer 2: ```suggestion
Generally, it seems that websites are starting to hear the call of users to respect their privacy (a call that is getting louder and louder). More and more sites are switching to employing browser features that restrict the information that is sent to third parties. Furthermore, mainly motivated by privacy regulations such as GDPR and CCPA, we are seeing a clear increase (of almost 60%) in the adoption of consent management platforms (CMPs), giving users more control over which information they want to share.
```

Finally, on the side of the browsers, we are also seeing a strong evolution towards providing users with more control of their online privacy. Next to the features that several privacy-focused browsers offer as a built-in solution, there is also the Privacy Sandbox initiative that aims to continue providing the current functionalities on the web (such as targeted advertising, anti-fraud, attribution of purchases, ...) without the nefarious side-effects of cross-site tracking. Although the development is still in fairly early stages, we see that web services on a substantial number of websites are already opting-in to the Origin Trial. As such, the features are extensively being tested, and are likely to become a persistent part of the web.

> While it may still take a couple of years to finally get there, we are transitioning towards a web that gives users more control over what they want to share with which parties. We can see this convergence on both sides of the spectrum: on the one hand initiated by the website, and on the other hand enforced by the browser. We can be hopeful that in the not-so-distant future the data we share, is the data that we intend to share, and the journey on the web that we take on a day-to-day basis no longer needs to be collected, shared and analyzed by the numerous trackers that we currently encounter—in the hope of respectfully tomorrow for all.

Reviewer 2: ```suggestion
While it may still take a couple of years to finally get there, we are transitioning towards a web that gives users more control over what they want to share with which parties. We can see this convergence on both sides of the spectrum: on the one hand initiated by the website, and on the other hand enforced by the browser. We can be hopeful that in the not-so-distant future the data we share, is the data that we intend to share, and the journey on the web that we take on a day-to-day basis no longer needs to be collected, shared, and analyzed by the numerous trackers that we currently encounter—in the hope of respectfully tomorrow for all.
```

nrllh_bio: Nurullah Demir is a cyber security researcher and PhD student at <a hreflang="en" href="https://www.internet-sicherheit.de/en/">Institute for Internet Security</a> and <a hreflang="en" href="https://intellisec.de">Intelligent System Security, KASTEL Security Research Labs</a>. His research focuses on web security & privacy, and web measurements.
results: https://docs.google.com/spreadsheets/d/1iJqj3g0VEjpmjzvtX6VLeRehE7LDQGcw6lOadxGxkjk/
featured_quote: While it may still take a couple of years to finally get there, we are transitioning towards a web that gives users more control over what they want to share with which parties. We can see this convergence on both sides of the spectrum&colon; on the one hand, initiated by the website, and on the other hand, enforced by the browser.
featured_stat_1: 82%
featured_stat_label_1: Sites including at least one third-party tracker.
featured_stat_2: 11%
featured_stat_label_2: The prevalence of Consent Management Platforms (CMPs).
featured_stat_3: 9.5%
featured_stat_label_3: Sites of the top 1,000 that make use of User-Agent Client Hints.

> unedited: true

Reviewer 3: ```suggestion
```
